### PR TITLE
Support Python 3.10 on S3

### LINKED ISSUE #1054: Support 3.10
- Seems we have issues with S3 plugin with 3.10

Need all CI to pass with 3.10 ... today it does not.

### PR DESCRIPTION
**Pull request by isidentical**
Fixes #1054

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -86,7 +86,7 @@ uvloop =
     uvloop
 
 s3 =
-    s3path
+    s3path==0.3.3
```
Comment (cooperlees) [inline setup.cfg:89]: I try not a version packages here so we don't lock people.
- Lets at least
```suggestion
    s3path>=0.3.3
```

**Review by cooperlees — CHANGES_REQUESTED** [review]
Awesome. Thanks for this! Appreciate it.

Can we also add a 3.10 classifier to setup.cfg please to state we support 3.10 + fix the s3 extra dependency versioning and I'll merge.

File: setup.cfg (hunk 1 above)
Comment (isidentical) [reply setup.cfg:89]: Upps, my bad. I thought I did `>=`. Thanks!

File: requirements_s3.txt [hunk 2]
```diff
@@ -2,7 +2,7 @@ boto3==1.20.17
 botocore==1.23.17
 jmespath==0.10.0
 python-dateutil==2.8.2
-s3path==0.3.2
+s3path>=0.3.3
```
Comment (cooperlees) [inline requirements_s3.txt:5]:
```suggestion
s3path==0.3.3
```
Here we pin to say this is the versions CI has tested.

**Review by cooperlees — APPROVED** [review]
Will fix before release.

File: CHANGES.md [hunk 3]
```diff
@@ -1,3 +1,9 @@
+# Unreleased
+
+## Other
+
+- 3.10 is now supported `PR #1073` -- Thanks **isidentical**
```
Comment (ichard26) [inline CHANGES.md:5]:
```suggestion
- Python 3.10 is now supported `PR #1073` -- Thanks **isidentical**
```

**Review by ichard26 — APPROVED** [review]

**Comment by ichard26** [comment]
Thank you for helping out @isidentical on so many projects in Cooper and I's world :slightly_smiling_face:

**Pull request merged**